Adds information about role policy and roleARN for OpenID docs. (#983)
minio/docs: source/administration/identity-access-management/oidc-access-management.rst

1. Use the `JSON Web Token claim <https://datatracker.ietf.org/doc/html/rfc7519#section-4>`__ returned as part of the OIDC authentication flow to identify the :ref:`policies <minio-policy>` to assign to the authenticated user.
2. Use the ``RoleARN`` specified in the authorization request to assign the policies attached to the provider's Role Policy.
   

MinIO by default denies access to all actions or resources not explicitly allowed by a user's assigned or inherited :ref:`policies <minio-policy>`. 
Users managed by an OIDC provider must specify the necessary policies as part of the JWT claim. If the user JWT claim has no matching MinIO policies, that user has no permissions to access any action or resource on the MinIO deployment.

The specific claim which MinIO looks for is configured as part of :ref:`deploying the cluster with OIDC identity management <minio-external-iam-oidc>`. This page focuses on creating MinIO policies to match the configured OIDC claims.

Authentication and Authorization Flow 
-------------------------------------

> A MinIO cluster can have any number of Role Policy based OpenID providers configured.

Reviewer: Suggest we intro lead and do the micro-toc here

Spitballing:

```
MinIO supports two OIDC authentication and authorization flows:

1. The RolePolicy flow sets the assigned policies for an authenticated user in the MinIO configuration.

2. The JWT flow sets the assigned policies for an authenticated user as part of the OIDC configuration.

MinIO supports **one** JWT claim-based OIDC provider per deployment. All other providers must use Role Policy.

Author: Applied this suggestion with some small tweaks.

A MinIO Cluster may only have **one** JWT Claim based OpenID provider configured.
Other providers must be configured with the Role Policy method instead.

Role Policy and RoleARN
~~~~~~~~~~~~~~~~~~~~~~~

MinIO recommends using the Role Policy method for authenticating with an OpenID provider.

> With a Role Policy, all registered users receive the same policy or policies for access.

Reviewer: ```suggestion
With a Role Policy, all clients which generate an STS credential using a given RoleARN receive the :ref:`policy or policies <minio-policy>` associated with that Role Policy configuration.
```

registered users receive the same policy or policies for access.
```

Reviewer: Maybe the flow here should be

- Use Role Policies to create an OIDC configuration with a specific set of assigned MinIO Policies

- When you create the configuration, MinIO generates and returns a `RoleARN` for use with the AssumeRoleWithWebIdentity STS API

- Any client who generates credentials using that `RoleArn` and API use those policies for access.

Author: Tweaking the suggestion and then updated the flow steps.

> Use :ref:`OpenID Policy Variables <minio-policy-variables-oidc>` to manage what each individual user has access to.

Reviewer: ```suggestion
You can use  :ref:`OpenID Policy Variables <minio-policy-variables-oidc>` to create policies that programmatically manage what each individual user has access to.
```

Author: 👍

   Authentication using Implicit Flow is not supported.

2. Specify the ``RoleARN`` to the MinIO Security Token Service (STS) :ref:`minio-sts-assumerolewithwebidentity` API endpoint. 
   
   .. note::

      MinIO assigns and displays a unique RoleARN to each configured OpenID provider at startup.

   MinIO verifies the ``RoleARN`` and checks for a :ref:`role policy <minio-external-identity-management-openid-access-control>` specifying a list
   of one or more :ref:`policies <minio-policy>` to assign to the authenticated user. 

   Use either the :envvar:`MINIO_IDENTITY_OPENID_ROLE_POLICY` environment variable or the :mc-conf:`identity_openid role_policy <identity_openid.role_policy>` configuration setting to define the list of policies to use for the provider

3. MinIO returns temporary credentials in the STS API response in the form of an access key, secret key, and session token. 
   The credentials have permissions matching those policies specified in the Role Policy.
   
4. Applications use the temporary credentials returned by the STS endpoint to perform authenticated S3 operations on MinIO.

JSON Web Token Claim
~~~~~~~~~~~~~~~~~~~~

> Using JSON Web Tokens allows for individual assignment of policies at the increased cost of managing multiple policies for separate claims.

Reviewer 2: This feels like it wants a comma?
```suggestion
Using JSON Web Tokens allows for individual assignment of policies, at the increased cost of managing multiple policies for separate claims.
```

Author: This actually doesn't call for a comma. But let me see if I can break it up some.
